### PR TITLE
Live weather input validation

### PR DESCRIPTION
**Pull request by PikaDude**
The Live Weather Location inputs now validate
the values by sending a test request to the API.
If there is an error, it does not save the inputs
and shows the error message to the user. If it
succeeds, then it shows the returned weather
status and location to the user before saving.

### CODE REVIEW

File: scripts/options.js [hunk 1]
```diff
@@ -118,6 +117,58 @@ onClickElements.forEach(el => {
...
+	if (zip == '') {
+		responseMessage('You must specify a zip code.');
+		return;
+	}
+	if (country == '') {
+		responseMessage('You must specify a country code.');
+		return;
+	}
+
+	let appid = "e7f97bd1900b94491d3263f89cbe28d6";
```
Comment (paladique) [inline scripts/options.js:136]: May not be a big deal and definitely not a blocker, but we could probably hide this key. Or, have a "production" key.

**Review by paladique — APPROVED** [review]
Looks good!

File: scripts/options.js (hunk 1 above)
Comment (PikaDude) [reply scripts/options.js:136]: Yeah, I'll work on a solution to that next. This key's been bundled with this extension for quite some time.

**Comment by paladique** [comment]
Just testing on my PC before merging!

**Pull request merged**